tmdb/client.py

```diff
@@ -92,8 +92,15 @@ def retrieve_show(self, show_id: int) -> Show:
         :param show_id: id of the show in the API.
         :return: a Show object
         """
-        resp = self._request(f'tv/{show_id}')
-        data: dict = resp.json()
+        show_resp = self._request(f'tv/{show_id}')
+        data: dict = show_resp.json()
+        season_count: int = data['number_of_seasons']
```

Comment:
here I access the data without a parser, is that okay ?

Reply:
Yes, it seems okay because otherwise the parser would need to know how to make HTTP requests… #SingleResponsibilityPrinciple.

However it would be a good idea to move all the season-fetching logic to a separate private method, for better clarity. :-)